theano/sandbox/gpuarray/basic_ops.py

```diff
@@ -923,13 +923,24 @@ def __hash__(self):
         return hash(self.dtype) ^ hash(type(self))
 
     def gpu_kernels(self, node, name):
-        code = """
+        if self.dtype == 'float16':
+            code = """
+KERNEL void k(GLOBAL_MEM %(ctype)s *a, ga_size n, ga_size m) {
+    ga_size nb = n < m ? n : m;
+    for (ga_size i = LID_0; i < nb; i += LDIM_0) {
+        a[i*m + i] = __float2half_rn(1);
```

Comment:
You could use `fp16_help.load_w` here, instead of duplicating the whole kernel, right?